src/ripple/app/misc/impl/ValidatorSite.cpp

```diff
@@ -455,6 +478,52 @@ ValidatorSite::onSiteFetch(
     cv_.notify_all();
 }
 
+void
+ValidatorSite::onTextFetch(
+    boost::system::error_code const& ec,
+    std::string&& res,
+    std::size_t siteIdx)
+{
+    bool shouldRetry = false;
+    {
+        std::lock_guard <std::mutex> lock_sites{sites_mutex_};
+        try
+        {
+            if (ec)
+            {
+                JLOG (j_.warn()) <<
+                    "Problem retrieving from " <<
+                    sites_[siteIdx].activeResource->uri <<
+                    " " <<
+                    ec.value() <<
+                    ":" <<
+                    ec.message();
+                shouldRetry = true;
+                throw std::runtime_error{"fetch error"};
```

Comment:
After reading the other discussions, I'm a little less sure, but it does make sense since there aren't as many "not a real problem" things that can go wrong with reading a file.